File: code/Android_App/ITS-Mobile-App/app/src/main/java/pt/isel/meic/tfm/itsmobileapp/security_protocols/dlapp/models/HashChain.kt
package pt.isel.meic.tfm.itsmobileapp.security_protocols.dlapp.models

/**
 * Class representing an HashChain
 *
 * A hash chain consists in a set of secret keys to perform a Message Authentication Code (MAC)
 */
class HashChain(
    val hashChain: Array<ByteArray>
) {
    override fun toString(): String {
        val sb = StringBuilder()

        sb.append("HashChain(hashChain=").append("\n")
        for (bytes in hashChain)
            sb.append(bytes.contentToString()).append("\n")
        sb.append(")")

        return sb.toString()
    }
}

File: code/MRSU_App/security_protocols/dlapp/models/hash_chain.py
class HashChain:
    """
    Class representing an hash chain
    """

    def __init__(self, hash_chain):
        self.chain = hash_chain

    def __str__(self):
        representation = "HashChain(hashChain="
        representation += "\n"
        for hash_bytes in self.chain:
            representation += str(hash_bytes) + "\n"
        representation += ")"

        return representation
